src/sources/datadog/logs.rs

```diff
@@ -0,0 +1,291 @@
+use crate::{
+    config::{
+        log_schema, DataType, GenerateConfig, GlobalOptions, Resource, SourceConfig,
+        SourceDescription,
+    },
+    event::Event,
+    shutdown::ShutdownSignal,
+    sources::{
+        self,
+        http::{decode_body, Encoding},
+        util::{ErrorMessage, HttpSource, HttpSourceAuthConfig},
+    },
+    tls::TlsConfig,
+    Pipeline,
+};
+use bytes::Bytes;
+use lazy_static::lazy_static;
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+use std::{collections::HashMap, net::SocketAddr};
+
+use warp::http::HeaderMap;
+
+lazy_static! {
+    static ref API_KEY_MATCHER: Regex =
+        Regex::new(r"^/v1/input/(?P<api_key>[[:alnum:]]{32})/??").unwrap();
```

Comment:
Not a big deal, but I would have expected us to parse the URL instead of using a regex. What would you think of doing that instead via `std::path`?

Reply:
I did a quick attempt with `std::path`, then I think about windows platform, then I checked the rust doc https://doc.rust-lang.org/src/std/path.rs.html#255-259 and I end up believing (not 100% sure) that windows platform will probably have trouble parsing `/` separated token with `std::path`. That's mostly why I moved to a platform independent solution.

Reply:
👍 that's a fair point. I still feel like it'd be better to do URL parsing (maybe at a layer above to pass the path as a &[str]), but the regex is sufficient for now.